### PR TITLE
Fix some Code Inspections issues

### PR DESCRIPTION
**Pull request by freddyboucher**
Code Style issues:
- 'size() == 0' replaceable with 'isEmpty()'
- Type parameter explicitly extends 'java.lang.Object'
- Unnecessary 'this' qualifier
- Unnecessary enum modifier
- Unnecessary fully qualified name
- Unnecessary interface modifier
- Unnecessary semicolon

Data flow issues:
- Redundant local variable

Imports:
- Unused import

Inheritance issues:
- Missing @Override annotation

Java 5:
- 'for' loop replaceable with 'foreach'
- 'indexOf()' expression is replaceable with 'contains()'
- 'StringBuffer' may be 'StringBuilder'
- Unnecessary unboxing

Probable bugs:
- Unused assignment

Performance issues:
- Instantiating object to get Class object
- Manual array to collection copy
- Use bulk operation instead of iteration

Verbose or redundant code constructs:
- Redundant array creation
- Redundant type arguments
- Redundant type cast

@mkristian 
This PR contains #352 and #353 so don't review it before #352 and #353 have been merged (otherwise you won't see the proper diff).
In other words when you review this PR, only this commit https://github.com/freddyboucher/resty-gwt/commit/b8181219a7a7654ccf858472eb5a8c59112ce36e should be present.
Thanks

### CODE REVIEW

**Comment by mkristian** [comment]
@freddyboucher would it OK for you if I add you as contributor ? I am happy to review all your PR even then. but all these formatting, build related stuff is also OK for you if you merge them right away.

**Comment by freddyboucher** [comment]
@mkristian 
Oh yes thank you.
Of course I will still open PR for non cosmetic changes.
